File: 2.linked-list/remove-duplicate/removeDuplicate.js
// For sorted linkedlist
let removeDuplicate1 = function (head) {
  if (!head) {
    return head;
  }
  let prev = head;
  let current = head.next;
  while (current) {
    if (current.val !== prev.val) {
      prev.next = current;
      prev = current;
    }
    current = current.next;
  }
  prev.next = null;
  return head;
};

File: 2.linked-list/single-linked-list/remove-duplicate/remove-duplicate.rb
def remove_duplicate(head)
  if head.nil?
    return head
  end
  prev = head
  current = head.next
  while current do
    if current.val != prev.val
      prev.next = current
      prev = current
    end
    current = current.next
  end
  prev.next = nil
  return head
end